tests/perf-test/SearchPage.perf-test.js

```diff
@@ -0,0 +1,232 @@
+import ONYXKEYS from '../../src/ONYXKEYS';
+import createCollection from '../utils/collections/createCollection';
+import createPersonalDetails from '../utils/collections/personalDetails';
+import createRandomReport from '../utils/collections/reports';
+import PusherHelper from '../utils/PusherHelper';
+import * as TestHelper from '../utils/TestHelper';
+import waitForBatchedUpdates from '../utils/waitForBatchedUpdates';
+import wrapOnyxWithWaitForBatchedUpdates from '../utils/wrapOnyxWithWaitForBatchedUpdates';
+
+jest.mock('../../src/libs/Navigation/Navigation');
+
+const mockedNavigate = jest.fn();
+jest.mock('@react-navigation/native', () => {
+    const actualNav = jest.requireActual('@react-navigation/native');
+    return {
+        ...actualNav,
+        useFocusEffect: jest.fn(),
+        useIsFocused: () => ({
+            navigate: mockedNavigate,
+        }),
+        useRoute: () => jest.fn(),
+        useNavigation: () => ({
+            navigate: jest.fn(),
+            addListener: () => jest.fn(),
+        }),
+        createNavigationContainerRef: jest.fn(),
+    };
+});
+
+const getMockedReports = (length = 100) =>
+    createCollection(
+        (item) => `${ONYXKEYS.COLLECTION.REPORT}${item.reportID}`,
+        (index) => createRandomReport(index),
+        length,
+    );
+
+const getMockedPersonalDetails = (length = 100) =>
+    createCollection(
+        (item) => item.accountID,
+        (index) => createPersonalDetails(index),
+        length,
+    );
+
+const mockedReports = getMockedReports(600);
+const mockedBetas = _.values(CONST.BETAS);
+const mockedPersonalDetails = getMockedPersonalDetails(100);
+
+beforeAll(() =>
+    Onyx.init({
+        keys: ONYXKEYS,
+        safeEvictionKeys: [ONYXKEYS.COLLECTION.REPORT],
+        registerStorageEventListener: () => {},
+    }),
+);
+
+// Initialize the network key for OfflineWithFeedback
+beforeEach(() => {
+    global.fetch = TestHelper.getGlobalFetchMock();
+    wrapOnyxWithWaitForBatchedUpdates(Onyx);
+    Onyx.merge(ONYXKEYS.NETWORK, {isOffline: false});
+});
+
+// Clear out Onyx after each test so that each test starts with a clean state
+afterEach(() => {
+    Onyx.clear();
+    PusherHelper.teardown();
+});
+
+function SearchPageWrapper(args) {
+    return (
+        <ComposeProviders components={[OnyxProvider, CurrentReportIDContextProvider, KeyboardStateProvider, WindowDimensionsProvider, LocaleContextProvider]}>
+            <SearchPage
+                // eslint-disable-next-line react/jsx-props-no-spreading
+                {...args}
+                navigation={args.navigation}
+            />
+        </ComposeProviders>
+    );
+}
+
+const runs = CONST.PERFORMANCE_TESTS.RUNS;
+
+/**
+ * This is a helper function to create a mock for the addListener function of the react-navigation library.
+ * Same approach as in ReportScreen.perf-test.js
+ *
+ * P.S: This can't be moved to a utils file because Jest wants any external function to stay in the scope.
```

Comment:
Is there really no way to share such methods across Jest tests?

Reply:
I moved it to `TestHelper` file and tests worked properly :)